stripe/src/main/java/com/stripe/android/StripePaymentController.kt

```diff
@@ -471,28 +486,36 @@ internal class StripePaymentController internal constructor(
         )
     }
 
+    private suspend fun authenticateAlipay(
+        paymentIntent: PaymentIntent,
+        authenticator: AlipayAuthenticator,
+        requestOptions: ApiRequest.Options
+    ): PaymentIntentResult {
+        val outcome =
+            alipayRepository.authenticate(paymentIntent, authenticator, requestOptions).outcome
+        val refreshedPaymentIntent = requireNotNull(
+            stripeRepository.retrievePaymentIntent(
+                paymentIntent.clientSecret.orEmpty(),
+                requestOptions,
+                expandFields = EXPAND_PAYMENT_METHOD
+            )
+        )
+        return PaymentIntentResult(
+            refreshedPaymentIntent,
+            outcome,
+            failureMessageFactory.create(refreshedPaymentIntent, outcome)
+        )
+    }
+
     @VisibleForTesting
-    internal suspend fun authenticateAlipay(
+    internal suspend fun authenticateAlipayWithCallback(
```

Comment:
Can we use the same pattern we used where we'd push the callback to `Stripe` and not have to pass it into `StripePaymentController`?

Reply:
Sure, now `StripePaymentController` is callback free 😃